lib/src/main/scala/spinal/lib/bus/regif/Block/FifoInst.scala

```diff
@@ -0,0 +1,40 @@
+package spinal.lib.bus.regif
+
+import spinal.core._
+import spinal.lib.Stream
+
+abstract class FifoInst(name: String, addr: BigInt, doc: String, grp: GrpTag = null)(bi: BusIf) extends RegSlice(name, addr, doc, size = bi.wordAddressInc, grp)(bi){
+  override val regType: String = "FIFO"
+
+  val hitDoRead: Bool
+  val hitDoWrite : Bool
+
+  val bus = Stream(Bits(bi.busDataWidth bit))
```

Comment:
Having a `Stream` here creates an issue for `WrFifoInst` where it's lying to the user: Stream implies that backpressure can be applied, which is not the case there. So for write this should be a `Flow` instead. This would correctly indicate that it can't stall the bus, and the user can use `toStream(overflow: Bool)` to convert to a stream with an overflow check (or use `toStream` if they don't care).

In short: this should be moved to `WrFifoInst` (as a `Flow`) and `RdFifoInst` (kept as a `Stream`)

Reply:
make sense ,  `Flow` is better than `Stream` for WrFfifo